src/libsync/logger.cpp

```diff
@@ -272,44 +282,49 @@ void Logger::enterNextLogFile()
-
-        // Expire old log files and deal with conflicts
-        const auto &files = dir.entryList(QStringList(QStringLiteral("*owncloud.log.*")),
-            QDir::Files, QDir::Name);
-        QRegExp rx(QStringLiteral(R"(.*owncloud\.log\.(\d+).*)"));
-        int maxNumber = -1;
-        for (const auto &s : files) {
-            if (_logExpire.count() > 0) {
-                std::chrono::seconds expireSeconds(_logExpire);
-                QFileInfo fileInfo(dir.absoluteFilePath(s));
-                if (fileInfo.lastModified().addSecs(expireSeconds.count()) < now) {
-                    dir.remove(s);
-                }
-            }
-            if (s.startsWith(newLogName) && rx.exactMatch(s)) {
-                maxNumber = qMax(maxNumber, rx.cap(1).toInt());
-            }
+        const QString logName = dir.filePath(QStringLiteral(APPLICATION_SHORTNAME ".log"));
+        QString previousLog;
+
+        if (_logFile.isOpen()) {
+            _logFile.close();
         }
-        newLogName.append(QLatin1Char('.') + QString::number(maxNumber + 1));
-
-        auto previousLog = _logFile.fileName();
-        setLogFile(dir.filePath(newLogName));
-
-        // Compress the previous log file. On a restart this can be the most recent
-        // log file.
-        auto logToCompress = previousLog;
-        if (logToCompress.isEmpty() && files.size() > 0 && !files.last().endsWith(QLatin1String(".gz")))
-            logToCompress = dir.absoluteFilePath(files.last());
-        if (!logToCompress.isEmpty()) {
-            QString compressedName = logToCompress + QStringLiteral(".gz");
-            if (compressLog(logToCompress, compressedName)) {
-                QFile::remove(logToCompress);
-            } else {
-                QFile::remove(compressedName);
+        // rename previous log file if size != 0
+        const auto info = QFileInfo(logName);
+        if (info.exists(logName) && !info.size() == 0) {
+            previousLog = dir.filePath(QStringLiteral(APPLICATION_SHORTNAME "-%1.log").arg(info.created().toString(QStringLiteral("MMdd_hh.mm.ss.zzz"))));
+            if (!QFile(logName).rename(previousLog)) {
+                std::cerr << "Failed to rename: " << qPrintable(logName) << " to " << qPrintable(previousLog) << std::endl;
             }
         }
+
+        const auto now = QDateTime::currentDateTime();
+        open(logName);
+        // set the creation time to now
+        _logFile.setFileTime(now, QFileDevice::FileTime::FileBirthTime);
+
+        QtConcurrent::run([now, previousLog, dir, logExpire = _logExpire] {
+            // Expire old log files and deal with conflicts
+            const auto &files = dir.entryList(QStringList(QStringLiteral("*" APPLICATION_SHORTNAME "-*.log.gz")),
+                QDir::Files, QDir::Name);
+            for (const auto &s : files) {
+                if (logExpire.count() > 0) {
+                    std::chrono::seconds expireSeconds(logExpire);
```

Comment:
Now that the amount of log files is quite low, deleting everything older than 4h feels dangerous, we might only have one or two in that time range.

I'd rather go for the more common approach of keeping up to N logs?

Reply:
Sounds like a good plan. However, 100 MiB of logs is a lot of data. Maybe lower the number to 10 MiB and keep more files? Concatenating them is trivial.